packages/bpk-component-banner-alert/src/BpkBannerAlert.tsx

```diff
@@ -15,21 +15,19 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* @flow strict */
-import {
-  CommonProps,
-  COMMON_PROP_TYPES,
-  COMMON_DEFAULT_PROPS,
-} from './common-types';
+
+import type { CommonProps } from './common-types';
 import BpkBannerAlertInner from './BpkBannerAlertInner';
 
-type Props = {
-  ...$Exact<CommonProps>,
+const defaultProps = {
+  animateOnEnter: false,
+  animateOnLeave: false,
+  show: true,
+  icon: null,
 };
 
-const BpkBannerAlert = (props: Props) => <BpkBannerAlertInner {...props} />;
-
-BpkBannerAlert.propTypes = COMMON_PROP_TYPES;
-BpkBannerAlert.defaultProps = COMMON_DEFAULT_PROPS;
+const BpkBannerAlert = (props: CommonProps) => (
+  <BpkBannerAlertInner {...defaultProps} {...props} />
+);
```

Comment:
Should we be passing through the `defaultProps` here instead of defining them as we have in other components as part of the props: CommonProps definition?

Only reason I question is because I have a worry that due to some weirdness that React could do at runtime if someone looks to override these default values it could accidently be they get applied after the consumer `props` values so causing a potential bug of being applied in the wrong order and not giving the expected outcome

Reply:
oh interesting, is that like a known bug or something? the order in which props are passed should definitely matter. I only wrote it this way because didn't want to have to pass each prop individually given it's just prop drilling and nothing in addition